testing/src/main/java/org/oppia/android/testing/TextInputAction.kt

```diff
@@ -0,0 +1,50 @@
+package org.oppia.android.testing
+
+import android.content.Context
+import android.view.View
+import androidx.annotation.StringRes
+import com.google.android.material.textfield.TextInputLayout
+import org.hamcrest.Description
+import org.hamcrest.Matcher
+import org.hamcrest.TypeSafeMatcher
+import javax.inject.Inject
+
+/**
+ * Action for matching error text and performing other actions on TextInputLayout in a test infrastructure-specific way.
+ */
+class TextInputAction @Inject constructor(val context: Context) {
+  // TODO(#1720): Move this to a companion object & use a test-only singleton injector to retrieve.
+
+  /**
+   * @param expectedErrorTextId is the string id which needs to match.
+   *
+   * @returns a [Matcher] that matches the specified string [expectedErrorTextId] with the view's error text.
+   */
+  fun hasErrorText(@StringRes expectedErrorTextId: Int): Matcher<View> {
```

Comment:
Alluded to in my comment above, but I suggest naming the parameter ``expectedErrorTextResId`` to emphasize this is a resource ID.

Reply:
Did the asked changes